packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -592,11 +834,160 @@ export default {
     setUserFiltered() {
       this.isUserFiltered = true;
     },
+
+    resetFilters() {
+      this.filters.rootObjects = [];
+      Object.keys(this.filters.declutter).forEach((k) => {
+        this.filters.declutter[k].on = this.filters.declutter[k].default;
+      });
+      this.filters.declutter.hideName.names = [];
+    },
+
+    addHiddenName(name) {
+      const objectName = name.trim();
+
+      if (
+        !objectName ||
+        this.filters.declutter.hideName.names.includes(objectName)
+      ) {
+        return;
+      }
+
+      this.filters.declutter.hideName.names.push(objectName);
+      this.filters.declutter.hideName.on = true;
+      this.clearSelection();
+    },
+
+    removeHiddenName(index) {
+      this.filters.declutter.hideName.names.splice(index, 1);
+      if (this.filters.declutter.hideName.names.length === 0) {
+        this.filters.declutter.hideName.on = false;
+      }
+      this.clearSelection();
+    },
+
+    addRootObject(fqid) {
+      const objectFqid = fqid.trim();
+
+      if (!objectFqid || this.filters.rootObjects.includes(objectFqid)) {
+        return;
+      }
+
+      this.filters.rootObjects.push(objectFqid);
+      this.clearSelection();
+    },
+
+    removeRootObject(index) {
+      this.filters.rootObjects.splice(index, 1);
+      this.clearSelection();
+    },
+
+    filterMediaRequests(events) {
+      const excludedEvents = [];
+      const mediaRegex = [
+        'application/javascript',
+        'application/ecmascript',
+        'audio/.+',
+        'font/.+',
+        'image/.+',
+        'text/javascript',
+        'text/ecmascript',
+        'text/css',
+        'video/.+',
+      ].map((t) => new RegExp(t, 'i'));
+      const mediaFileExtensions = [
+        'aac',
+        'avi',
+        'bmp',
+        'css',
+        'flv',
+        'gif',
+        'htm',
+        'html',
+        'ico',
+        'jpeg',
+        'jpg',
+        'js',
+        'json',
+        'jsonld',
+        'mid',
+        'midi',
+        'mjs',
+        'mov',
+        'mp3',
+        'mp4',
+        'mpeg',
+        'oga',
+        'ogg',
+        'ogv',
+        'ogx',
+        'opus',
+        'otf',
+        'png',
+        'svg',
+        'tif',
+        'tiff',
+        'ts',
+        'ttf',
+        'wav',
+        'weba',
+        'webm',
+        'webp',
+        'woff',
+        'woff2',
+        'xhtml',
+        '3gp',
+        '3g2',
+      ];
+
+      events.forEach((e) => {
+        const eventName = e.codeObject.prettyName;
+
+        if (
+          eventName.startsWith('GET') &&
+          mediaFileExtensions.some((ext) => eventName.endsWith(ext))
+        ) {
+          excludedEvents.push(e.id);
+        } else if (
+          e.http_server_response &&
+          e.http_server_response.mime_type &&
```

Comment:
Also check the `Content-Type` header if `mime_type` is undefined. `httpServerResponse.headers` may be undefined and the keys may vary in casing. Something like this should work (but, beware, I've not actually run this code):

```js
const [, contentType] = Object.entries(e.httpServerResponse.headers || {}).find(([k,v]) => k.toLowerCase() === 'content-type')
```

Reply:
@dustinbyrne  According to AppMap specification `httpServerResponse` doesn't have `headers` attribute:
https://github.com/applandinc/appmap#http-server-request-return-attributes

Reply:
It looks like that needs to be updated. `headers` is available as an optional parameter, preferred over `mime_type`.